common/persistence/cassandra/mutable_state_store.go

```diff
@@ -478,16 +478,38 @@ func (d *MutableStateStore) CreateWorkflowExecution(
 	return &p.InternalCreateWorkflowExecutionResponse{}, nil
 }
 
+// GetWorkflowExecution gets the workflow execution data from the cassandra DB.
+// If the request contains a runID it loads and returns the execution identified by that run ID.
+// If the request does not contain a runID, it first looks up the current run ID and then loads the corresponding execution object.
 func (d *MutableStateStore) GetWorkflowExecution(
 	ctx context.Context,
 	request *p.GetWorkflowExecutionRequest,
 ) (*p.InternalGetWorkflowExecutionResponse, error) {
+	runID := request.RunID
+	if runID == "" {
+		// get current runid using the permanentRunID record.
+		query := d.Session.Query(templateGetCurrentExecutionQuery,
```

Comment:
we need to change persistence rate limit client to use 2 tokens for this API when runID is not specified.

Reply:
Not sure I follow what 2 tokens means. Can you give some more context?